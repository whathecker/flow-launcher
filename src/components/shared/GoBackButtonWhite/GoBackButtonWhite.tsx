import React from "react";
import { Touchable } from "../../Themed";
import { Image, StyleSheet } from "react-native";

import { Container } from "../../../styles";

type GoBackButtonProps = {
  pressHandler: () => void;
  backgroundColor: string;
};

const GoBackButtonWhite: React.FC<GoBackButtonProps> = ({
  pressHandler,
  backgroundColor,
}: GoBackButtonProps) => {
  return (
    <Touchable
      style={{
        ...Container.centerAligned,
        backgroundColor: backgroundColor,
      }}
      onPress={pressHandler}
    >
      <Image
        style={styles.goBackIcon}
        source={require(`../../../../assets/images/arrow_white.png`)}
      />
    </Touchable>
  );
};

const styles = StyleSheet.create({
  goBackIcon: {
    width: 20,
    height: 20,
  },
});

export default GoBackButtonWhite;
