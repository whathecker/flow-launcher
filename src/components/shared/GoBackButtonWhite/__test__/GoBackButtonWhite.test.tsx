import * as React from "react";
import renderer from "react-test-renderer";
import GoBackButtonWhite from "../GoBackButtonWhite";

describe("Test GoBackButtonWhite component", () => {
  it("Should renders GoBackButtonWhite correctly", () => {
    const tree = renderer
      .create(
        <GoBackButtonWhite
          pressHandler={() => {
            return;
          }}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
