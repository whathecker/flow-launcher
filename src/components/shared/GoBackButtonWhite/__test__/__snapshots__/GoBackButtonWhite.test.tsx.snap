// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GoBackButtonWhite component Should renders GoBackButtonWhite correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": undefined,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 20,
        "width": 20,
      }
    }
  />
</View>
`;
